### PR TITLE
chore: Cleaned up README

### PR DESCRIPTION
**Pull request by rhuss**
This is just a start for reorganizing the client side documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
 
-The goals of the Knative Client are to:
+`kn` offers you ...
 
-1. Follow the Knative [serving](https://github.com/knative/serving) and [eventing](https://github.com/knative/eventing) APIs
-2. Be scriptable to allow users to create different Knative workflows
-3. Expose useful Golang packages to allow integration into other programs or CLIs or plugins
-4. Use consistent verbs, nouns, and flags for various commands
-5. Be easily extended via a plugin mechanism (similar to `kubectl`) to allow for experimentation and customization
+* ... full support for managing all features of [Knative Serving](https://github.com/knative/serving) (services, revisions, traffic splits)
```
Comment (daisy-ycguo) [inline README.md]: Some `Serving` start with an uppercase letter, some do not. Please use a uniform way. So does the word `Eventing`

Comment (rhuss) [reply README.md]: So the convention I used here was

* "Knative Serving" and "Knative Eventing" when referring to thos top-level components of "Knative" as a product/project
* `Service`, `Revision`, `Route` when referring to resources in a literal sense and concrete context.
* "services", "revisions"m "trafifc splits" when used in a more abstract and figurative meaning like here. The boundary to the resources is fluent so.

Does this make sense ?

Comment (daisy-ycguo) [reply README.md]: Yes. Thank you, Roland

File: README.md [hunk 2]
```diff
@@ -1,42 +1,36 @@
...
+* ... growing support [Knative eventing](https://github.com/knative/eventing), closely following its development (managing of sources & triggers)
+* ... a plugin architecture similar to that of `kubectl` plugins
+* ... a thin client-specific API in golang which helps in tasks like synchronously waiting on Knative service write operations.
+* ... easy integration of Knative into Tekton Pipelines by using `kn` in a Tekton `Task`.
```
Comment (navidshaikh) [inline README.md]: add a link for task https://github.com/tektoncd/catalog/tree/master/kn ?

File: README.md [hunk 3]
```diff
@@ -1,42 +1,36 @@
...
 
-**Shell auto completion:**
+Start with the [user's guide](docs/README.md) to learn more. You can read about common use cases, get detailed documentation on each command, and learn how to extend the `kn` CLI. For more information, have a look at:
 
-Run the following command to enable shell auto-completion:
+* [User guide](docs/README.md)
+  - Installation - How to install `kn` and run on your machine
+  - Examples - Use case based examples
+  - FAQ (_to come._)
+* [Reference Manual](docs/cmd/kn.md) - all possible commands and options with usage exampls
```
Comment (navidshaikh) [inline README.md]:
```suggestion
* [Reference Manual](docs/cmd/kn.md) - all possible commands and options with usage examples
```

File: README.md [hunk 4]
```diff
@@ -1,42 +1,36 @@
...
+* ... a plugin architecture similar to that of `kubectl` plugins
+* ... a thin client-specific API in golang which helps in tasks like synchronously waiting on Knative service write operations.
+* ... easy integration of Knative into Tekton Pipelines by using `kn` in a Tekton `Task`.
 
-# Docs
 
-Start with the [user's guide](docs/README.md) to learn more. You can read about common use cases, get detailed documentation on each command, and learn how to extend the `kn` CLI. For more information, access the following links:
+This client uses the [Knative Serving](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md) and [Knative Eventing](https://github.com/knative/eventing/tree/master/docs/spec) API exclusively so that it will work with any Knative installation, even those that are not Kubernetes based.
+It does not help in *installing* Knative itself though.
+Please refer to the various [Knative installation options](https://knative.dev/docs/install/) for how to Install Knative with its prerequisites.
```
Comment (navidshaikh) [inline README.md]:
```suggestion
Please refer to the various [Knative installation options](https://knative.dev/docs/install/) for how to install Knative with its prerequisites.
```

File: README.md [hunk 5]
```diff
@@ -1,42 +1,36 @@
...
 
-For Bash:
-```sh
-$ source <(kn completion bash)
-```
+We love contributions! Please refer to
+[CONTRIBUTING](https://knative.dev/contributing/) for more information on how to best contributed to contribute to Knative.
 
-Use TAB to list available sub-commands or flags.
+For code contributions it as easy as picking an [issue](https://github.com/knative/client/issues) (look out for "kind/good-first-issue"), briefly comment that you would like to work on it, code, code, code and finally submit a [PR](https://github.com/knative/client/pulls) which will trigger the review process.
```
Comment (maximilien) [inline README.md]: I’d mention: “code test code” :)

File: README.md (hunk 2 above)
Comment (maximilien) [reply README.md]: Not too sure about the `...` at start of the list. Not clear that makes it easier to harder to read. Anyhow my .02

**Comment by dsimansk** [comment]
Looks good, thanks!
/lgtm

**Pull request merged**